packages/file-upload/src/from-url/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { useState } from 'react';
+import { Box, TextField, Typography } from '@mui/material';
+import { FromURLContainer } from './styles';
+import { Button } from '../../../button/src/index';
+
+export interface FromUrlProps {
+  accept?: string;
+  acceptLabels?: string;
+  variant?: 'single' | 'combined';
+  componentDisabled?: boolean;
+  buttonDisabled?: boolean;
+  inputDisabled?: boolean;
+  onAccept?: (url: string) => void;
+  onReject?: (url: string) => void;
+  validator?: (url: string) => string | null;
+}
+
+export function FromUrl(props: FromUrlProps) {
+  const {
+    accept,
+    acceptLabels,
+    variant = 'single',
+    componentDisabled,
+    inputDisabled,
+    buttonDisabled,
+    onAccept = () => undefined,
+    onReject = () => undefined,
+    validator,
+  } = props;
+
+  const [url, setUrl] = useState('');
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  function isValidUrl(string) {
+    try {
+      new URL(string);
+      return true;
+    } catch (err) {
+      return false;
+    }
+  }
+
+  const handleValidation = async (url: string): Promise<string | null> => {
+    if (validator) return validator(url);
+    if (!url) return 'URL is required';
+    if (!isValidUrl(url)) return 'Invalid URL';
+    if (!accept) return null;
+
+    setLoading(true);
+    return fetch(url)
+      .then((response) => {
+        const type = response.headers.get('Content-Type');
+        // How to validate types as audio/*, video/*, image/*?
+        setLoading(false);
+        if (type && accept.includes(type)) {
+          return null;
+        } else {
+          return 'Invalid file type';
+        }
+      })
+      .catch(() => {
+        setLoading(false);
+        return 'Validation error';
+      });
+  };
+
+  const handleSubmit = async () => {
+    setError(null);
+    const validationResult = await handleValidation(url);
+    if (validationResult === null) {
+      onAccept(url);
+      setUrl('');
+    } else {
+      onReject(url);
```

Comment:
Should I add onError as well? If so, where to call it?

Reply:
I think in several places

1. Call when the download function is called and the error happens.
2. Call when handleValidate return false